### PR TITLE
[PUBDEV-7347] implement leaf node assignments for XGBoost

### PR DESCRIPTION
**Pull request by honzasterba**
- do the same for IsoFor while we're at it

### CODE REVIEW

File: h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java [hunk 1]
```diff
@@ -100,6 +100,21 @@ public boolean requiresOffset() {
...
+  }
+
+  @Override
+  public int getNTreesPerGroup() {
+    return _nclasses > 2 ? _nclasses : 1;
+  }
+
+  @Override
+  public String[] getDecisionPath(double[] row) {
+    return new String[0];
```
Comment (Pscheidl) [inline h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java]: Is this correct ?

Comment (honzasterba) [reply h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostMojoModel.java]: no! good catch!

File: h2o-genmodel/src/test/java/hex/genmodel/easy/EasyPredictModelWrapperTest.java [hunk 2]
```diff
@@ -190,7 +187,6 @@ public void testSerializeWrapper() throws Exception {
     SupervisedModel rawModel = makeSupervisedModel();
     EasyPredictModelWrapper m = new EasyPredictModelWrapper(rawModel);
 
-    Assert.assertTrue(m instanceof Serializable);
```
Comment (michalkurka) [inline h2o-genmodel/src/test/java/hex/genmodel/easy/EasyPredictModelWrapperTest.java]: does it cause any issue?

Comment (honzasterba) [reply h2o-genmodel/src/test/java/hex/genmodel/easy/EasyPredictModelWrapperTest.java]: its always true

Comment (michalkurka) [reply h2o-genmodel/src/test/java/hex/genmodel/easy/EasyPredictModelWrapperTest.java]: ok

**Pull request merged**